doc/utarray.txt

```diff
@@ -370,4 +370,12 @@ Notes
    For this reason, it's usually better to refer to an element by its integer
    'index' in code whose duration may include element insertion.
 
+6. To override the default out-of-memory handling behavior (which calls `exit(-1)`),
+   override the `utarray_oom` macro before including `utarray.h` or functions that use it.
+   For example,
+
+  #define utarray_oom() do { set_error_code(100); } while (0)
```

Comment:
Notice that if `utarray_oom()` is defined to something that does NOT either terminate execution or `longjmp` away from the trouble zone, then `utarray_new` will dereference a null pointer and segfault.

I will adjust this documentation to reflect that.

But, knowing that, does `utarray_oom` still solve your underlying problem? Or is it the wrong tool for the job?

In particular, should you instead be looking for a way to redefine the meaning of `malloc` in this code, so that it can't fail in the first place? uthash also provides `uthash_malloc` and `uthash_free`.

Reply:
Yes, it solves my problem. Thanks for asking.